### PR TITLE
Migrate to chart dependency for upstream grafana chart

### PR DESCRIPTION
**Pull request by QuantumEnigmaa**
towards https://github.com/giantswarm/giantswarm/issues/24518

:warning: DO NOT MERGE :warning: - Need to prepare the update of config repo's custom values and to synchronize release grafana and config.

### CODE REVIEW

**Comment by QuentinBisson** [comment]
Can you regerenate the schema?

File: .circleci/config.yml
Comment (QuentinBisson) [inline .circleci/config.yml:26]: I'd like it if we added this tbh

File: .circleci/ct-config.yaml
Comment (QuentinBisson) [inline .circleci/ct-config.yaml:3]:
```suggestion
- grafana=https://grafana.github.io/helm-charts

```

File: .gitignore
Comment (QuentinBisson) [inline .gitignore:7]:
```suggestion
helm/grafana/charts/*.tgz

```

File: helm/grafana/Chart.yaml
Comment (QuentinBisson) [inline helm/grafana/Chart.yaml:3]:
```suggestion
version: 2.0.0
```

**Comment by QuentinBisson** [comment]
Can you remove the binary in the charts/ directory?

File: CHANGELOG.md
Comment (TheoBrigitte) [inline CHANGELOG.md:18]: This is an internal CI change, I would not mention this in the changelog.

File: .circleci/config.yml
Comment (TheoBrigitte) [inline .circleci/config.yml:11]: Can we solves the error instead of changing the executor ?

File: .circleci/ct-config.yaml
Comment (TheoBrigitte) [inline .circleci/ct-config.yaml:3]: New line at EOF is missing

File: .gitignore
Comment (TheoBrigitte) [inline .gitignore:7]: New line at EOF is missing

File: helm/grafana/templates/_helpers.tpl
Comment (QuentinBisson) [reply helm/grafana/templates/_helpers.tpl:14]: That will solve the abs issue

File: helm/grafana/Chart.yaml
Comment (TheoBrigitte) [inline helm/grafana/Chart.yaml:3]: I guess this goes along with changing the executor in the CI. Could we fix the error and keep the version here ?

**Comment by QuentinBisson** [comment]
Can you post a diff between the old template and the new template usign `helm template helm/grafana` before and after the change?

File: helm/grafana/Chart.yaml
Comment (TheoBrigitte) [inline helm/grafana/Chart.yaml:17]: Maintainers is Giantswarm Team Atlas
```suggestion
  - name: giantswarm/team-atlas
    email: team-atlas@giantswarm.io
```

**Review by TheoBrigitte — CHANGES_REQUESTED** [review]
Nice job. But let's try to fix CI if we can.

File: .circleci/config.yml
Comment (hervenicol) [reply .circleci/config.yml:26]: Requires changes in app-build-suite. Today, all our "chart-depending" apps have had to remove this.
I agree though, would be good to update app-build-suite now we know it's the way we're going with our apps.

File: helm/grafana/Chart.yaml
Comment (hervenicol) [reply helm/grafana/Chart.yaml:3]: Current CI fails if we have the wrong `version`, and suggests we set it to `"[[ .Version ]]`.

File: helm/grafana/templates/_helpers.tpl
Comment (hervenicol) [reply helm/grafana/templates/_helpers.tpl:14]: This would not set the right team label because, in the sub-chart, `application.giantswarm.io/team` does not exist / is empty.

File: helm/grafana/Chart.yaml
Comment (hervenicol) [reply helm/grafana/Chart.yaml:17]: We took it from loki-app IIRC, so we should update it there too :grin:

**Comment by hervenicol** [comment]
Also, added "do not merge" mention in PR description. Better safe than sorry :grin:

File: helm/grafana/Chart.yaml
Comment (TheoBrigitte) [reply helm/grafana/Chart.yaml:17]: Here you go https://github.com/giantswarm/loki-app/pull/126

**Comment by QuantumEnigmaa** [comment]
I updated the values.schema.json file as well as the CHANGELOG and the Chart.yaml, removed the binary from the `charts` folder and added a new line at EOF for each specified file.

File: .circleci/config.yml
Comment (QuentinBisson) [reply .circleci/config.yml:26]: I added it to the prometheus-operator-app without issues though. Why do we need to remove it?

File: helm/grafana/Chart.yaml
Comment (QuentinBisson) [reply helm/grafana/Chart.yaml:3]: If we use architect yes, with ABS we need to set the version manually

File: helm/grafana/templates/_helpers.tpl
Comment (QuentinBisson) [reply helm/grafana/templates/_helpers.tpl:14]: This is working for the promethues-operator-app so I;m a bit perplexed

Comment (QuentinBisson) [reply helm/grafana/templates/_helpers.tpl:14]: Actually it was not :( how did I miss it ..

Comment (QuentinBisson) [reply helm/grafana/templates/_helpers.tpl:14]: This should solve it https://github.com/giantswarm/app-build-suite/pull/234

File: .abs/main.yaml
Comment (QuentinBisson) [inline .abs/main.yaml:6]: Can you add a comment explaining why? Otherwise looks good to me :)

Comment (QuentinBisson) [reply .abs/main.yaml:6]: Otherwise LGTM

**Review by QuentinBisson — APPROVED** [review]
LGTM, I'd like some other eyes though :)

File: helm/grafana/templates/_helpers.tpl
Comment (QuantumEnigmaa) [reply helm/grafana/templates/_helpers.tpl]: I updated the label with the annotation you suggested

File: helm/grafana/Chart.yaml
Comment (QuantumEnigmaa) [reply helm/grafana/Chart.yaml]: I updated the Chart.yaml with this change

Comment (QuantumEnigmaa) [reply helm/grafana/Chart.yaml]: As I mentioned with Quentin, I have now put manually version: 2.0.0 there. Therefore we can keep the version here.

**Review by TheoBrigitte — APPROVED** [review]
LGTM, please make sure to test deploy this both fresh install and via upgrade.

**Comment by hervenicol** [comment]
Tested on a MC (on `ginger`), using config from https://github.com/giantswarm/config/pull/1406 and app from https://github.com/giantswarm/grafana-app/pull/74.

Looks good:
* grafana deployed
* dex authentication
* dashboards are there
* datasource is OK

**Comment by QuantumEnigmaa** [comment]
Tested on a WC (also on `ginger`), using happa app installation tab.

Encountered AppArmor issue mentioned here : https://github.com/giantswarm/giantswarm/issues/24677 
Apart for this, looks good : 
- grafana deployed 
- authentication working

**Comment by QuentinBisson** [comment]
Awesome work <3

**Review by hervenicol — APPROVED** [review]
I think now we're good for release :+1:

**Pull request merged**
